src/typing_extensions.py

```diff
@@ -846,6 +849,43 @@ def __round__(self, ndigits: int = 0) -> T_co:
             pass
 
 
+if hasattr(io, "Reader") and hasattr(io, "Writer"):
+    Reader = io.Reader
+    Writer = io.Writer
+else:
+    @runtime_checkable
+    class Reader(Protocol[T_co]):
+        """Protocol for simple I/O reader instances.
+
+        This protocol only supports blocking I/O.
+        """
+
+        __slots__ = ()
+
+        @abc.abstractmethod
+        def read(self, size: int = ..., /) -> T_co:
+            """Read data from the input stream and return it.
+
+            If size is specified, at most size items (bytes/characters) will be
+            read.
+            """
+
+    @runtime_checkable
+    class Writer(Protocol[T_contra]):
+        """Protocol for simple I/O writer instances.
+
+        This protocol only supports blocking I/O.
+        """
+
+        __slots__ = ()
+
+        @abc.abstractmethod
+        def write(self, data: T_contra, /) -> int:
+            """Write data to the output stream and return the number of items written."""
+
+
+
+
 _NEEDS_SINGLETONMETA = (
```

Comment:
there's maybe a few too many empty lines here :P

```suggestion
        def write(self, data: T_contra, /) -> int:
            """Write data to the output stream and return the number of items written."""


_NEEDS_SINGLETONMETA = (
```